### PR TITLE
Cycle.js Converter

### PR DESCRIPTION
**Pull request by manuelroemer**
# New Pull Request

## Prerequisites

Please make sure you can check the following boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] All new and existing tests passed

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [x] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes

### Description

This PR adds a Cycle.js converter (see #232). A new package `piral-cycle` has been introduced for this.

For reference: Here is a modified version of the CrossFx demo which shows an (in-place) integration of this converter: https://gist.github.com/manuelroemer/40fd358c3eb487c6c52e9f449b684b84

### Remarks

Feel free to comment on the changes and suggest improvements! Especially on the naming and the exposed types.

### CODE REVIEW

File: src/converters/piral-cycle/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  // ...
+});
+```
+
+The `@cycle/run`, `@cycle/dom` and `xstream` packages should be shared with the pilets via the *package.json*:
+
+```json
+{
+  "pilets": {
+    "externals": [
```
Comment (manuelroemer) [inline src/converters/piral-cycle/README.md:86]: @FlorianRappl Is this right, i.e. should the piral-cycle dependencies be listed here?

Comment (FlorianRappl) [reply src/converters/piral-cycle/README.md:86]: Yeah makes sense!

File: src/converters/piral-cycle/README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
+[![Piral Logo](https://github.com/smapiot/piral/raw/master/docs/assets/logo.png)](https://piral.io)
```
Comment (FlorianRappl) [inline src/converters/piral-cycle/README.md:1]: Does Cycle.js have a nice logo?

We could add it to `src/pages/docs/src/assets/extensions` as a png (cycle.png).

Comment (manuelroemer) [reply src/converters/piral-cycle/README.md:1]: There is this one: https://cycle.js.org/img/cyclejs_logo.svg
We'd just have to convert it to .png.

Comment (manuelroemer) [reply src/converters/piral-cycle/README.md:1]: Alright, done!

**Pull request merged**